pilot/pkg/model/push_context.go

```diff
@@ -259,18 +259,34 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 		Reason: append(first.Reason, other.Reason...),
 	}
 
-	// Only merge EdsUpdates when incremental eds push needed.
-	if !merged.Full {
-		merged.EdsUpdates = make(map[string]struct{})
-		// Merge the updates
-		for update := range first.EdsUpdates {
-			merged.EdsUpdates[update] = struct{}{}
+	// Do not merge when any one is empty
+	if len(first.ConfigsUpdated) > 0 && len(other.ConfigsUpdated) > 0 {
+		merged.ConfigsUpdated = make(map[resource.GroupVersionKind]map[string]struct{})
+		for kind := range first.ConfigsUpdated {
+			merged.ConfigsUpdated[kind] = make(map[string]struct{})
 		}
-		for update := range other.EdsUpdates {
-			merged.EdsUpdates[update] = struct{}{}
+		for kind := range other.ConfigsUpdated {
+			merged.ConfigsUpdated[kind] = make(map[string]struct{})
+		}
+
+		if !merged.Full {
```

Comment:
This to merge all kinds of specific configs if only not full push, currently happens only on incremental eds. 

If so, i am wondering why not move this check to the front as previously

Reply:
The first part merges the config types changed. The latter merges the changed resources like incremental eds does. To have a better understanding, you can check the "simple merge" unit test case in "push_context_test.go".

Reply:
sounds good